### PR TITLE
Restructure Developer Guide to Fit Standards

### PR DESCRIPTION
**Pull request by mynameizzhafeez**

### CODE REVIEW

**Review by zichen-3974 — APPROVED** [review]
LGTM. UiClassDiagram outdated after recent GUI update, but can edit in future too since we planned for more changes in v1.3. 

Some future considerations: 
- Do you think we can consider dropdown lists for the content and links, like the textbook? I don't think conversion to pdf will pose a problem if we do it ourselves, unless it's getting auto-converted by the marking side, then maybe not.
- Since images need to be put in a table for captioning, should we make the border transparent? Likely will have to use html table. 
- Should some image explanations be on the side instead? See below, the guide feels off balance to me, I think we could either centre all images or move the explanation to beside the image (possible problem with this is it might be inconsistent with images that can not have explanations on the side due to their size). Also needs html likely.
image

**Comment by mynameizzhafeez** [comment]
Yeah it makes sense to shove the images to the center.

**Comment by zichen-3974** [comment]
I think <div> with align would work too. Another thing is do you think there are to many emojis? I get that more can make it more specialised but would it be too much to remember since we wont want them to keep scrolling back to what each emoji means?

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -2,41 +2,276 @@
...
+## **Using the Guide**
+### **What Do the Emojis Mean**
+
+| Emoji                | GFMD form              | What does it mean                           |
+|----------------------|------------------------|---------------------------------------------|
+| :sparkles:           | `:sparkles:`           | Completed feature                           |
+| :construction:       | `:construction:`       | Feature under construction                  |
+| :question:           | `:question:`           | Frequently asked question (FAQ)             |
+| :white_check_mark:   | `:white_check_mark:`   | What the developer should do                |
+| :exclamation:        | `:exclamation:`        | Warning to the developer                    |
```
Comment (seanfirefox) [inline docs/DeveloperGuide.md:111]: suggestion to change to `:warning:` :warning: instead

**Review by seanfirefox — APPROVED** [review]
LGTM!

**Pull request merged**